src/Streamly/Data/Fold.hs

```diff
@@ -6,61 +6,55 @@
 -- Stability   : experimental
 -- Portability : GHC
 --
--- 'Fold' type represents an effectful action that consumes a value from an
--- input stream and combines it with a single final value often called an
--- accumulator, returning the resulting output accumulator.  Values from a
--- stream can be /pushed/ to the fold and consumed one at a time. It can also
--- be called a consumer of stream or a sink.  It is a data representation of
--- the standard 'Streamly.Prelude.foldl'' function.  A 'Fold' can be turned
--- into an effect (@m b@) using 'Streamly.Prelude.fold' by supplying it the
--- input stream.
+-- The 'Fold' type represents an effectful action that absorbs a value into an
+-- accumulator. For example, a 'sum' 'Fold' represents adding the input to the
+-- accumulated sum.  A fold driver e.g. 'Streamly.Prelude.fold' pushes values
+-- from a stream to the 'Fold' one at a time, reducing the stream to a single
+-- value. Therefore, a fold can also be thought of as a sink or a stream
+-- consumer.
 --
--- Using this representation multiple folds can be combined efficiently using
--- combinators; a stream can then be supplied to the combined fold and it would
--- distribute the input to constituent folds according to the composition.  For
--- example, an applicative composition distributes the same input to the
--- constituent folds and then combines the resulting fold outputs.  Similarly,
--- a partitioning combinator divides the input among constituent folds.
+-- A 'Fold' is in fact a data representation of a standard left fold ('foldl').
+-- Unlike the standard left folds, 'Fold's can terminate at any point e.g. the
+-- 'head' fold would terminate immediately after returning the head element.
 --
--- = Monoids
+-- 'Fold's can be combined efficiently using combinators provided in this
+-- module; a stream supplied to the combined fold is distributed to the
```

Comment:
`distributed` might not be the right word to use. Consider splitting.

Reply:
The word is used in a general layman sense. I can change it to "suppllied".

Reply:
Changed to "provided".